lib/collection/src/collection/collection_ops.rs

```diff
@@ -212,11 +211,13 @@ impl Collection {
 
                 // ...and cancel transfer tasks and remove transfers from internal state
                 for transfer in transfers {
-                    self.finish_shard_transfer(transfer, Some(&shard_holder))
+                    self.abort_shard_transfer(transfer.key(), Some(&shard_holder))
                         .await?;
                 }
             }
 
+            replica_set.remove_peer(peer_id).await?;
```

Comment:
It feels safer to remove shard after the transfer is aborted

Reply:
This happens if the "drop_replica" operation is requested. In this PR we don't introduce new logic of removing shard, we simply change form removing shard before cancelling transfers to after.

Reply:
I see now. Thanks!